openwisp_controller/config/tests/utils.py

```diff
@@ -213,6 +217,35 @@ def _create_vpn(self, ca_options={}, **kwargs):
 
 
 class CreateConfigTemplateMixin(CreateTemplateMixin, CreateConfigMixin):
+    class _CustomAssertnumQueriesContext(_AssertNumQueriesContext):
+        def __exit__(self, exc_type, exc_value, traceback):
+            if exc_type is not None:
+                return
+            if 'openwisp_controller.subnet_division' in getattr(
+                settings, 'INSTALLED_APPS'
+            ):
+                for query in self.captured_queries:
+                    if 'subnetdivision' in query['sql']:
+                        self.num += 1
+            super().__exit__(exc_type, exc_value, traceback)
+
+    def assertNumQueries(self, num, func=None, *args, using=DEFAULT_DB_ALIAS, **kwargs):
+        # "openwisp_controller.subnet_division" updates context of
+        # the Config object which creates additional database queries.
+        # Usage of subnet_division app is optional hence, tests in
+        # "openwisp_controller.config" are written assuming
+        # subnet_division is not used. But when it is used, the number
+        # of queries should be increased which is done by
+        # _CustomAssertnumQueriesContext.__exit__ method.
+        conn = connections[using]
```

Comment:
This part can be moved to subnet_divisions app. I decided to keep it here in config to keep code saner. 

If we choose to move this to subnet_division, we will need to mokeypatch `CreateConfigTemplateMixin`.

Reply:
I have already moved it because the tests for sample app were failing.